src/Transactions/TransactionDetails/TransactionDetails.js

```diff
@@ -58,14 +58,17 @@ const TransactionDetails = ({
             amount={transaction.amount}
             currency={transaction.currency}
             description={transaction.description}
+            encumbrance={transaction.encumbrance}
             fiscalYearCode={fiscalYearCode}
+            fiscalYearId={transaction.fiscalYearId}
             fromFundName={fromFundName}
+            invoiceId={transaction.sourceInvoiceId}
+            invoiceLineId={transaction.sourceInvoiceLineId}
             metadata={transaction.metadata}
```

Comment:
IMHO, it's anti-pattern. If all purpose of TransactionDetails.js to show details of transaction object - it's better to pass just the whole object instead of making new props on every schema change or new ui requirement

Reply:
primitives are better than objects

Reply:
any proofs?) I provided pros from objects in props for such cases. Of course we are talking that object is immutable.

Reply:
I've seen somewhere it's the best practice, but I don't remember where, mb in my dreams :)

with primitives we shouldn't worry about object immutability

Reply:
Let's pass the whole object in this case